server/src/main/java/org/elasticsearch/script/ScriptCache.java

```diff
@@ -121,24 +109,28 @@
                     }
                     // Check whether too many compilations have happened
                     checkCompilationLimit();
```

Comment:
The call to `checkCompilationLimit()` is no longer locked on the `ScriptCache`, we'll have to make this call safe.

Reply:
Yeah noticed that too that's why I added commit https://github.com/elastic/elasticsearch/pull/55467/commits/b55e8e38fbc8c54226ed62a468199a16eae89b19